src/libstd/collections/hash/map.rs

```diff
@@ -537,8 +538,8 @@ impl<K, V, S> HashMap<K, V, S>
                 Full(b) => b.into_bucket(),
             };
             buckets.next();
+            debug_assert!(buckets.index() < limit_bucket);
```

Comment:
This is an infinite loop if there is no free bucket, but that there is, is a requirement already in the caller (resize), so it seems ok.

(Note to other readers: buckets.index() keeps increasing even if the pointer it corresponds to wraps back to the first bucket.)